contribs/src/main/java/com/netflix/conductor/contribs/queue/kafka/KafkaObservableQueue.java

```diff
@@ -0,0 +1,230 @@
+
+  private static final String KAFKA_PREFIX = "kafka.";
+
+  private final String queueName;
+
+  private final int pollTimeInMS;
+
+  private final int longPollTimeout;
+
+  private final int pollCount;
+
+  private final KafkaProducer<String, String> producer;
+
+  private final KafkaConsumer<String, String> consumer;
+
+  @Inject
+  public KafkaObservableQueue(String queueName, Configuration config) {
+
+    this.queueName = queueName;
+    this.pollTimeInMS = config.getIntProperty("workflow.dyno.queues.pollingInterval", 100);
+    this.pollCount = config.getIntProperty("workflow.dyno.queues.pollCount", 10);
+    this.longPollTimeout = config.getIntProperty("workflow.dyno.queues.longPollTimeout", 1000);
+    // Init Kafka producer and consumer properties
+    Properties producerProps = new Properties();
+    Properties consumerProps = new Properties();
+    consumerProps.put("group.id", queueName + "_group");
+    String serverId = config.getServerId();
+    consumerProps.put("client.id", queueName + "_consumer_"+serverId);
+    producerProps.put("client.id", queueName + "_producer_"+serverId);
+    Map<String, Object> configMap = config.getAll();
+    for (Entry<String, Object> entry : configMap.entrySet()) {
+      String key = entry.getKey();
+      String value = (String) entry.getValue();
+      if (key.startsWith(KAFKA_PREFIX)) {
+        if (key.startsWith(KAFKA_PRODUCER_PREFIX)) {
+          producerProps.put(key.replaceAll(KAFKA_PRODUCER_PREFIX, ""), value);
+        } else if (key.startsWith(KAFKA_CONSUMER_PREFIX)) {
+          consumerProps.put(key.replaceAll(KAFKA_CONSUMER_PREFIX, ""), value);
+        } else {
+          producerProps.put(key.replaceAll(KAFKA_PREFIX, ""), value);
+          consumerProps.put(key.replaceAll(KAFKA_PREFIX, ""), value);
+        }
+      }
+    }
+
+    // Init Kafka producer and consumer
+    producer = new KafkaProducer<>(producerProps);
+    consumer = new KafkaConsumer<>(consumerProps);
+    consumer.subscribe(Collections.singletonList(queueName));
+    logger.info("KafkaObservableQueue initialized for {}", queueName);
+  }
+
+  @Override
+  public Observable<Message> observe() {
+    OnSubscribe<Message> subscriber = getOnSubscribe();
+    return Observable.create(subscriber);
+  }
+
+  @Override
+  public List<String> ack(List<Message> messages) {
+    return messages.stream().map(Message::getId).collect(Collectors.toList());
+  }
+
+  public void setUnackTimeout(Message message, long unackTimeout) {
+  }
+
+  @Override
+  public void publish(List<Message> messages) {
+    publishMessages(messages);
+  }
+
+  @Override
+  public long size() {
+    return 0;
+  }
+
+  @Override
+  public String getType() {
+    return QUEUE_TYPE;
+  }
+
+  @Override
+  public String getName() {
+    return queueName;
+  }
+
+  @Override
+  public String getURI() {
+    return queueName;
+  }
+
+  private List<Message> receiveMessages() {
+    List<Message> messages = new ArrayList<>();
+    try {
+
```

Comment:
If this method is not called within the time frame of a hear-beat being sent to the Kafka broker from the consumer, the consumer will be considered dead. This will lead to the consumer considered as dead and there will be no messages received.  What version of Kafka Cluster is this code targeted at ?

Reply:
Currently targetting the new version of Kafka which has inbuilt heartbeat support. Please suggest